mono/sgen/sgen-marksweep.c

```diff
@@ -693,6 +694,56 @@ major_alloc_object (GCVTable vtable, size_t size, gboolean has_references)
 }
 
 /*
+ * This can only be called by sgen workers. While this is called we assume
+ * that no other thread is accessing the block free lists. The world should
+ * be stopped and the gc thread should be waiting for workers to finish.
```

Comment:
Is it necessary for this code to work that the GC thread is waiting for workers to finish?